src/main/java/betterquesting/handlers/SaveLoadHandler.java

```diff
@@ -143,143 +217,139 @@ public void loadDatabases(MinecraftServer server) {
             JsonHelper.CopyPaste(fileLives, new File(BQ_Settings.curWorldDir + "/backup/" + fsVer, "LifeDatabase_backup_" + fsVer + ".json"));
         }
 
-        ILegacyLoader loader = LegacyLoaderRegistry.getLoader(fVer);
+        legacyLoader = LegacyLoaderRegistry.getLoader(formatVer);
 
-        if (loader == null) {
-            QuestSettings.INSTANCE.readFromNBT(nbt1.getCompoundTag("questSettings"));
-            QuestDatabase.INSTANCE.readFromNBT(nbt1.getTagList("questDatabase", 10), false);
-            QuestLineDatabase.INSTANCE.readFromNBT(nbt1.getTagList("questLines", 10), false);
+        if (legacyLoader == null) {
+            QuestSettings.INSTANCE.readFromNBT(nbt.getCompoundTag("questSettings"));
+            QuestDatabase.INSTANCE.readFromNBT(nbt.getTagList("questDatabase", 10), false);
+            QuestLineDatabase.INSTANCE.readFromNBT(nbt.getTagList("questLines", 10), false);
         } else {
-            loader.readFromJson(j1);
+            legacyLoader.readFromJson(json);
         }
 
         if (useDef) QuestSettings.INSTANCE.setProperty(NativeProps.EDIT_MODE, false); // Force edit off
         hasUpdate = packName.equals(QuestSettings.INSTANCE.getProperty(NativeProps.PACK_NAME)) && packVer > QuestSettings.INSTANCE.getProperty(NativeProps.PACK_VER);
+    }
 
-        // === PROGRESS ===
-
-        JsonObject j2 = JsonHelper.ReadFromFile(fileProgress);
+    private void loadProgress() {
+        if (fileProgress.exists()) {
+            JsonObject json = JsonHelper.ReadFromFile(fileProgress);
 
-        if (loader == null) {
-            NBTTagCompound nbt2 = NBTConverter.JSONtoNBT_Object(j2, new NBTTagCompound(), true);
-            QuestDatabase.INSTANCE.readProgressFromNBT(nbt2.getTagList("questProgress", 10), false);
-        } else {
-            loader.readProgressFromJson(j2);
+            if (legacyLoader == null) {
+                NBTTagCompound nbt = NBTConverter.JSONtoNBT_Object(json, new NBTTagCompound(), true);
+                QuestDatabase.INSTANCE.readProgressFromNBT(nbt.getTagList("questProgress", 10), false);
+            } else {
+                legacyLoader.readProgressFromJson(json);
+            }
         }
 
-        // === PARTIES ===
-
-        JsonObject j3 = JsonHelper.ReadFromFile(fileParties);
-
-        NBTTagCompound nbt3 = NBTConverter.JSONtoNBT_Object(j3, new NBTTagCompound(), true);
-        PartyManager.INSTANCE.readFromNBT(nbt3.getTagList("parties", 10), false);
-
-        // === NAMES ===
-
-        JsonObject j4 = JsonHelper.ReadFromFile(fileNames);
-
-        NBTTagCompound nbt4 = NBTConverter.JSONtoNBT_Object(j4, new NBTTagCompound(), true);
-        NameCache.INSTANCE.readFromNBT(nbt4.getTagList("nameCache", 10), false);
-
-        // === LIVES ===
+        getPlayerProgressFiles().forEach(file -> {
+            JsonObject json = JsonHelper.ReadFromFile(file);
```

Comment:
If there's any chance of an exception being thrown in this block it could result in a really ugly stack trace. I would honestly suggest just doing a standard for-each loop here.